src/mod/server.mod/servmsg.c

```diff
@@ -1286,51 +1425,71 @@ static int gotcap(char *from, char *msg) {
      * END. Future eggheads: add support for more than 1 of these async
      * capabilities, right now SASL is the only one so we're OK.
      */
-//    if (strstr(msg, "sasl") != NULL) {
-//      putlog(LOG_SERV, "*", "SASL AUTH CALL GOES HERE!");   //TODO
-//    }
-    if (!strstr(cap.negotiated, "sasl")) {
+    if (strstr(cap.negotiated, "sasl")) {
+      if ((sasl_mechanism != SASL_MECHANISM_ECDSA_NIST256P_CHALLENGE) 
+#ifdef HAVE_OPENSSL_SSL_H
+        || HAVE_OPENSSL_SSL_H
```

Comment:
Need to handle if sasl-mechanism is 1 (NIST256) but openssl is not on system- either alternate method, or CAP END